core/src/Kokkos_View.hpp

```diff
@@ -357,11 +357,123 @@ struct ViewTraits {
 template< class DataType , class ... Properties >
 class View ;
 
+template< class > struct is_view : public std::false_type {};
+
+template< class D, class ... P >
+struct is_view< View<D,P...> > : public std::true_type {};
+
+template< class D, class ... P >
+struct is_view< const View<D,P...> > : public std::true_type {};
+
+
 } /* namespace Kokkos */
 
 //----------------------------------------------------------------------------
 //----------------------------------------------------------------------------
 
+namespace Kokkos { namespace Impl {
```

Comment:
This section should appear after the View implementation, and then would not have to move the `is_view` fragment as well.